source/byte_buf.c

```diff
@@ -514,7 +513,7 @@ int aws_byte_buf_reserve(struct aws_byte_buf *buffer, size_t requested_capacity)
     buffer->buffer = new_buffer;
     buffer->capacity = requested_capacity;
 
-    AWS_FATAL_ASSERT(is_valid_byte_buf(buffer));
+    assert(is_valid_byte_buf(buffer));
```

Comment:
I'd squash this commit with the commit introducing the changes in the first place. Otherwise it makes for a messy git history.